test/test_sam.py

```diff
@@ -2149,3 +2151,137 @@ def testSoergelMatrixWithOneQueryInCommonExplicitReferenceIds(self):
                 matrix,
             )
         )
+
+
+@dataclass
+class AlignedRead:
+    """
+    Hold aligned read attributes that correspond to the pysam.AlignedRead class.
+
+    We cannot just make an instance of that class using pure Python because
+    it is implemented in C using data read from a BAM/SAM file and its attributes
+    are read-only.
+    """
+
+    is_duplicate: bool = False
+    is_qcfail: bool = False
+    is_secondary: bool = False
+    is_supplementary: bool = False
+    query_name: str = ""
+    reference_length: int = 0
+    reference_start: int = 0
+
+
+class TestReferenceReads(TestCase):
+    """
+    Test the ReferenceReads class.
+    """
+
+    def testAttributes(self):
+        "A ReferenceReads instance must have attributes with the expected values."
+        rr = ReferenceReads("id", 500)
+        self.assertEqual("id", rr.id_)
+        self.assertEqual(500, rr.length)
+        # The read id sets must all be empty.
+        for attr in (
+            "coveredOffsets",
+            "duplicate",
+            "nonDuplicate",
+            "primary",
+            "qcFail",
+            "readIds",
+            "secondary",
+            "supplementary",
+        ):
+            self.assertEqual(set(), getattr(rr, attr))
+
+    def testQueryName(self):
+        "Query names (read ids) must be added correctly."
```

Comment:
Don't we usually do 
```
"""
sometext
""" 

in situations like this?

Reply:
When it's just one short line I just use single quotes. There's a ton of such tests (in our code). Probably I should be more consistent...